Sources/XMLCoder/Auxiliaries/XMLCoderElement.swift

```diff
@@ -217,9 +217,16 @@ struct XMLCoderElement: Equatable {
         _ string: inout String,
         _ charactersEscapedInAttributes: [(String, String)]
     ) {
+        let actuallyAttributes = self.elements.filter {
+            $0.key.isEmpty && $0.elements.isEmpty && !$0.attributes.isEmpty
```

Comment:
is `key` a critical check here? Maybe your original case fails because of the key value intrinsic feature described here? https://github.com/MaxDesiatov/XMLCoder#coding-key-value-intrinsic